### PR TITLE
Update genbindings.sh and Cargo.toml for out-of-tree builds

### PR DESCRIPTION
**Pull request by TheBlueMatt**
\+ add out-of-tree builds.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
The CI run for this is at https://github.com/TheBlueMatt/ldk-c-bindings/runs/2059727595 which is looking good.

File: genbindings.sh [hunk 1]
```diff
@@ -3,6 +3,12 @@
 set -e
 set -x
 
+if [ ! -d "$1/lightning" ]; then
+	echo "USAGE: $0 path-to-rust-lightning"
+	exit 1
+fi
+LIGHTNING_PATH="$(realpath "$1/lightning")"
```
Comment (jkczyz) [inline genbindings.sh:10]: Looks like `realpath` doesn't exist on macos.

File: genbindings.sh [hunk 2]
```diff
@@ -3,6 +3,12 @@
 set -e
 set -x
 
+if [ ! -d "$1/lightning" ]; then
```
Comment (arik-so) [inline genbindings.sh:6]: this is good magic

File: genbindings.sh [hunk 3]
```diff
@@ -24,20 +30,27 @@ OUT="$(pwd)/lightning-c-bindings/src"
...
+if [ "$HOST_PLATFORM" = "host: x86_64-apple-darwin" ]; then
+	# OSX sed is for some reason not compatible with GNU sed
+	sed -i '' 's|lightning = { .*|lightning = { path = "'"$LIGHTNING_PATH"'" }|' lightning-c-bindings/Cargo.toml
+else
+	sed -i 's|lightning = { .*|lightning = { path = "'"$LIGHTNING_PATH"'" }|' lightning-c-bindings/Cargo.toml
+fi
 
 # Now cd to lightning-c-bindings, build the generated bindings, and call cbindgen to build a C header file
 PATH="$PATH:~/.cargo/bin"
 cd lightning-c-bindings
```
Comment (arik-so) [inline genbindings.sh:50]: why not use pushd here, too?

**Review by arik-so — APPROVED** [review]
LGTM, but curious about pushd

**Review by valentinewallace — APPROVED** [review]

File: genbindings.sh (hunk 3 above)
Comment (TheBlueMatt) [reply genbindings.sh:50]: I just used pushd earlier cause its easier than tracking the original `$PWD`

File: genbindings.sh (hunk 1 above)
Comment (TheBlueMatt) [reply genbindings.sh]: Pushed a version with `readlink` does that work? Otherwise may need a cd && `$PWD` tracker.

Comment (jkczyz) [reply genbindings.sh]: Getting:

```term
readlink: illegal option -- f
usage: readlink [-n] [file ...]
```

Comment (TheBlueMatt) [reply genbindings.sh]: Ok, I switched it to cd and resolve with pwd.

File: lightning-c-bindings/Cargo.toml [hunk 4]
```diff
@@ -17,7 +17,8 @@ crate-type = ["staticlib"
 [dependencies]
 bitcoin = "0.26"
 secp256k1 = { version = "0.20.1", features = ["global-context-less-secure"] }
-lightning = { version = "0.0.12", path = "../lightning" }
+# Note that the following line is matched by genbindings to update the path
+lightning = { git = "https://github.com/rust-bitcoin/rust-lightning", rev = "8b4ea56966878140b11aa3e17b44fcf1e01d6413" }
```
Comment (jkczyz) [inline lightning-c-bindings/Cargo.toml:21]: Does it matter what the target repo is at? I'm seeing:

```term
+ g++ -Wall -Wno-nullability-completeness -pthread -std=c++11 -Wall -g -pthread demo.cpp -Ltarget/debug/ -lldk -ldl
demo.cpp:511:46: error: no matching function for call to 'get_route'
                        LDK::CResult_RouteLightningErrorZ route = get_route(ChannelManager_get_our_node_id(&cm1), &graph_2_ref, ChannelManager_get_our_node_id(&cm2), &outbound_channels, LDKCVec_RouteHintZ {
                                                                  ^~~~~~~~~
./include/lightning.h:8976:40: note: candidate function not viable: requires 9 arguments, but 8 were provided
struct LDKCResult_RouteLightningErrorZ get_route(struct LDKPublicKey our_node_id, const struct LDKNetworkGraph *NONNULL_PTR network, struct LDKPublicKey payee, struct LDKInvoiceFeatures payee_features, struct LDKCVec_ChannelDetailsZ *first_hops, struct LDKCVec_RouteHintZ last_hops, uint64_t final_value_msat, ...
                                       ^
1 error generated.
```

Comment (jkczyz) [reply lightning-c-bindings/Cargo.toml:21]: Ah, maybe since that is pointing to a branch that hasn't been merged?

Comment (TheBlueMatt) [reply lightning-c-bindings/Cargo.toml:21]: Oh, oops, i must've accidentally ran it against the wrong branch. Let me merge the "drop C bindings" PR upstream and then rebuild.

Comment (jkczyz) [reply lightning-c-bindings/Cargo.toml:21]: I was actually running against your branch. When I run it against a branch that was merged, everything works.

**Pull request merged**